### PR TITLE
Refactor: update react-redux and redux-toolkit to latest versions

### PR DESCRIPTION
**Pull request by compojoom**
## What it solves
While I was tracking why a component re-renders I found out that it is due to a reducer always returning a new array. Those kind of errors should actually be reported in dev mode, but our version of redux-tookit was not doing that. After I quickly looked at the changelog it seemed that the relevant code for this was added in redux tookit version 5, but we were on 4. 

I had to bite the bullet and do the update. I did try to do this in the past, but back then didn't want to deal with the type problems this update was bringing with it... Now I no longer had a choice. :)

## Checklist
* [ ] I've tested the branch on mobile 📱
* [ ] I've documented how it affects the analytics (if at all) 📊
* [ ] I've written a unit/e2e test for it (if applicable) 🧑‍💻

### CODE REVIEW

File: src/store/broadcast.ts [hunk 1]
```diff
@@ -1,22 +1,23 @@
...
     // Broadcast actions that aren't being already broadcasted
-    if (!action._isBroadcasted) {
-      const sliceType = action.type.split('/')[0]
-      if (sliceNames.includes(sliceType)) {
-        broadcast?.postMessage({ action, tabId })
+    if (typeof action === 'object' && action !== null) {
+      const actionObj = action as { _isBroadcasted?: boolean; type?: string }
+      if (!actionObj._isBroadcasted && actionObj.type) {
+        const sliceType = actionObj.type.split('/')[0]
+        if (sliceNames.includes(sliceType as K)) {
```
Comment (katspaugh) [inline src/store/broadcast.ts:18]: Was it doing more broadcasts than needed before?

**Review by katspaugh — APPROVED** [review]
Looks good!

File: src/store/broadcast.ts (hunk 1 above)
Comment (compojoom) [reply src/store/broadcast.ts:18]: Those changes are more due to the new ts types. I had to silence ts. I don't think that there was an error in the previous code.

**Comment by katspaugh** [comment]
Let's merge it directly, since it's mostly mechanical changes.

**Pull request merged**